completions: bash: added a bunch of missing completion options.

lukester1975 opened the pull request:
#11422 was merged but 7 months old, so here's bash support for options added since then that can hopefully go in before the next release.

Perhaps a comment in `coredata.py` (or wherever best) alongside

```
# Update `docs/markdown/Builtin-options.md` after changing the options below
# Also update mesonlib._BUILTIN_NAMES. See the comment there for why this is required.
```

about updating shell completion helpers might help keep things synced?

Thanks

tristan957 commented on data/shell-completions/bash/meson:
```diff
@@ -195,6 +195,7 @@ _meson-setup() {
     sysconfdir
```
I wonder if there is some refactor here that could save us from duplicating everything

tristan957 commented:
Wanna throw in a commit which adds that comment?

lukester1975 commented:
> Wanna throw in a commit which adds that comment?

Turns out there is one here:

https://github.com/mesonbuild/meson/blob/da78b969ee3635bbea81909aa9ebdad3c77257a5/mesonbuild/mesonmain.py#L68-L69

lukester1975 commented:
Updated with the common setup/configure options factored out.

Note I removed `backend-startup-project` because AFAICT that just doesn't work from the command line, even when `backend=vs` is specified. It's possible I had patches from #9611 when doing the original fixes.

Which makes #9611 more important, as it seems the only way to use the feature is with `backend_startup_project` in `project(...)` which then fails unless the VS backend is used... any way to nudge it along?

Thanks

tristan957 commented:
> Though maybe one in coredata.py and each command's add_arguments() would help.

I think this makes sense. I added your other PR to the 1.3.0 milestone. Let's cross our fingers!

lukester1975 commented:
OK, added a note to all `add_arguments()`. Not that all commands have completion implemented in the bash helper yet, but maybe it'll help keep those that have in sync...

tristan957 commented:
you might want to split the added comments into their own commit.

lukester1975 commented:
OK, done.

jpakkane commented:
Test failure is unrelated.

bruchar1 commented:
I think this should be in 1.3.0 milestone, not 1.4.0...
